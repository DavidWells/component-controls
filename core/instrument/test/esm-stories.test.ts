import { loadStoriesTests } from './loadTestFiles';

describe('esm-stories', () => {
  loadStoriesTests(undefined, ['esm', 'stories']);
});
