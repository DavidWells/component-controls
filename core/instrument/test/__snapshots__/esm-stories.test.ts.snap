// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esm-stories named-object-export.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.503Z,
    "dateModified": 2020-10-28T15:50:29.503Z,
    "package": "8f81046c7a6e2fe97f79684cc333ec8a",
    "title": "Story",
  },
  "packages": Object {
    "8f81046c7a6e2fe97f79684cc333ec8a": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "8f81046c7a6e2fe97f79684cc333ec8a",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/named-object-export.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {},
  "transformed": "export default { title: 'Story' };
export const myStory = {
  i: 1,
};
",
}
`;

exports[`esm-stories no-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.503Z,
    "dateModified": 2020-10-28T15:50:29.503Z,
    "package": "1895a3a93b527a144ec98fabbd2fb747",
    "title": "Story",
  },
  "packages": Object {
    "1895a3a93b527a144ec98fabbd2fb747": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "1895a3a93b527a144ec98fabbd2fb747",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/no-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
",
}
`;

exports[`esm-stories no-story.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.503Z,
    "dateModified": 2020-10-28T15:50:29.504Z,
    "package": "c2ae0344d6ab23a6ef2c179448054b90",
    "title": "Story",
  },
  "packages": Object {
    "c2ae0344d6ab23a6ef2c179448054b90": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "c2ae0344d6ab23a6ef2c179448054b90",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/no-story.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {},
  "transformed": "export default { title: 'Story' };
const i = 1;
",
}
`;

exports[`esm-stories props-argument.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.504Z,
    "dateModified": 2020-10-28T15:50:29.504Z,
    "package": "cb9e6c0ae5060beae7c71ca34b71682f",
    "title": "Story",
  },
  "packages": Object {
    "cb9e6c0ae5060beae7c71ca34b71682f": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "cb9e6c0ae5060beae7c71ca34b71682f",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/props-argument.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 5,
              "line": 0,
            },
            "start": Object {
              "column": 0,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 34,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "props => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = props => {};
",
}
`;

exports[`esm-stories three-levels-alias.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.504Z,
    "dateModified": 2020-10-28T15:50:29.504Z,
    "package": "586922b358407a3990be308e1ca8b6a3",
    "title": "Story",
  },
  "packages": Object {
    "586922b358407a3990be308e1ca8b6a3": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "586922b358407a3990be308e1ca8b6a3",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/three-levels-alias.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 31,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 24,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "name",
              "value": Array [
                Object {
                  "loc": Object {
                    "end": Object {
                      "column": 16,
                      "line": 0,
                    },
                    "start": Object {
                      "column": 11,
                      "line": 0,
                    },
                  },
                  "name": "first",
                  "value": "first",
                },
                Object {
                  "loc": Object {
                    "end": Object {
                      "column": 22,
                      "line": 0,
                    },
                    "start": Object {
                      "column": 18,
                      "line": 0,
                    },
                  },
                  "name": "last",
                  "value": "last",
                },
              ],
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 29,
                  "line": 0,
                },
                "start": Object {
                  "column": 26,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 61,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name: { first, last }, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name: { first, last }, age }) => {};
",
}
`;

exports[`esm-stories two-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.504Z,
    "dateModified": 2020-10-28T15:50:29.504Z,
    "package": "90cb2d11a3e9b6dc7d194e3b67dd2ec8",
    "title": "Story",
  },
  "packages": Object {
    "90cb2d11a3e9b6dc7d194e3b67dd2ec8": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "90cb2d11a3e9b6dc7d194e3b67dd2ec8",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/two-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 6,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
        Object {
          "loc": Object {
            "end": Object {
              "column": 15,
              "line": 0,
            },
            "start": Object {
              "column": 8,
              "line": 0,
            },
          },
          "name": "context",
          "value": "context",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 45,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "(props, context) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = (props, context) => {};
",
}
`;

exports[`esm-stories two-levels-alias.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.504Z,
    "dateModified": 2020-10-28T15:50:29.504Z,
    "package": "ab364c24528da63d957c86821ba1c25f",
    "title": "Story",
  },
  "packages": Object {
    "ab364c24528da63d957c86821ba1c25f": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "ab364c24528da63d957c86821ba1c25f",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/two-levels-alias.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 21,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 14,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "name",
              "value": "MyNam",
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 19,
                  "line": 0,
                },
                "start": Object {
                  "column": 16,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 51,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name: MyNam, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name: MyNam, age }) => {};
",
}
`;

exports[`esm-stories two-levels-sub-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.504Z,
    "dateModified": 2020-10-28T15:50:29.504Z,
    "package": "e267c65a6808b78476145d626af8f521",
    "title": "Story",
  },
  "packages": Object {
    "e267c65a6808b78476145d626af8f521": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "e267c65a6808b78476145d626af8f521",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/two-levels-sub-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 14,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 7,
                  "line": 0,
                },
                "start": Object {
                  "column": 3,
                  "line": 0,
                },
              },
              "name": "name",
              "value": "name",
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 12,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 44,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name, age }) => {};
",
}
`;

exports[`esm-stories typescript.ts 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-10-28T15:50:29.505Z,
    "dateModified": 2020-10-28T15:50:29.505Z,
    "package": "deb7e98329844a351f2be938193836a3",
    "title": "Story",
  },
  "packages": Object {
    "deb7e98329844a351f2be938193836a3": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.36.6",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.21.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "deb7e98329844a351f2be938193836a3",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/typescript.ts",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.36.7",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 18,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 48,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "(props: Properties) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = (props: Properties) => {};
",
}
`;
